app/controllers/users/sessions_controller.rb

```diff
@@ -167,14 +167,25 @@ def handle_valid_authentication
       UserAlerts::AlertUserAboutNewDevice.schedule_alert(event:) if new_device?
       EmailAddress.update_last_sign_in_at_on_user_id_and_email(
         user_id: current_user.id,
-        email: auth_params[:email],
+        email: last_email_from_sp,
```

Comment:
Question: Do we still need the changes in this file? At a high level, I feel like if I've submitted my personal email address, we should be saving "last signed-in email" as my personal email address, even if what's shared with the partner is something different.

Have some of your recent changes allowed us to remove these changes? Assuming we're not dependent on the last signed-in email to know what email should be shared with the partner.